### ISSUE TITLE
Virus Total Detects Bundled File "541c6aa57ddd7da0c6902aa1e92155eb.virus"

### ISSUE THREAD

**Issue by SkyNinja** [PiKeyAr/sadx-mod-installer#7]
According to VirusTotal, the web installer contains a bundled file rather blatantly named "541c6aa57ddd7da0c6902aa1e92155eb.virus", which is detected as malicious by 51 vendors.

Interestingly, VT finds the bundled "SaveTransfer.exe", but not "7zr.exe"

541c6aa57ddd7da0c6902aa1e92155eb.virus was first submitted on 2022-03-27 05:36:43 UTC and has an apparent compile time of 2011-03-15 04:06:07 UTC.

[sadx_setup.exe VT page](https://www.virustotal.com/gui/file/c42d96bb0c3f2126d763dc14cc53aa1ae609892f468dd9bba80c09c172a1fadb/relations)
[541c6aa57ddd7da0c6902aa1e92155eb.virus VT Page](https://www.virustotal.com/gui/file/3cfa885d1a41e96f03a7261e29544da25da2f1138fcb4a1b295b2ccfcdee5c0d)

**Comment by PiKeyAr** [comment]
The [source code for SaveTransfer](https://github.com/PiKeyAr/sadx-mod-installer/tree/master/tools/SaveTransfer) is available on this repository. I'm not sure what you want me to do about this.

**Comment by SkyNinja** [comment]
Well, it shows right on VT that save transfer isn't the detected issue.
I just wanted you to know that something seems to have cropped up within the latest version, since the first time this thing was detected was within a month of last update to this repository.

**Comment by PiKeyAr** [comment]
I don't think it was the first time. Antiviruses have always had issues with the installer and its components, particularly the InetC NSIS plugin, which they think is malware. It's mentioned on the FAQ on the installer's website. I have no idea what `541c6aa57ddd7da0c6902aa1e92155eb.virus` is. The file `sadx_setup.exe` is produced by the NSIS compiler (I use Venis IX 2.2.5 as a visual environment).

VirusTotal detecting viruses or malware in the tools is a relatively common thing in general. You could try telling whoever is involved with these platforms about the false positive. Or, if you think there's an actual virus, you could try compiling your own version using the source code on this repo and comparing it with the EXE on the website.

**Comment by SkyNinja** [comment]
I don't think this is a false positive.

`541c6aa57ddd7da0c6902aa1e92155eb.virus` seems to [drop](https://www.virustotal.com/gui/file/3cfa885d1a41e96f03a7261e29544da25da2f1138fcb4a1b295b2ccfcdee5c0d/relations) and [execute](https://www.virustotal.com/gui/file/3cfa885d1a41e96f03a7261e29544da25da2f1138fcb4a1b295b2ccfcdee5c0d/behavior) various files, including what seems to be [an infected copy of Chocolatey](https://www.virustotal.com/gui/file/0ea4c0c3a9e6e646deadcd98b01a3d6cf0a07db9dcacd2d10a04e682cd1eafc2) which drops this [executable disguised as a changelog](https://www.virustotal.com/gui/file/3900b62cfc8f3d74e1b535ea8830778c9757191b2bc00c8f2c45848d286a9d1d/relations), and an executable called [Zombie.exe](https://www.virustotal.com/gui/file/a675a3cb3d2c68312ef35739c40a097c849e26ecc05fd09c565167f2e85c2e04/relations) which is dropped into the system folder and is also dropped by various other *.virus files.

**Comment by PiKeyAr** [comment]
Is there any evidence that any of this happens when the mod installer is used? I have just run a full installation with Process Monitor turned on, and I'm not seeing anything like that. Also, none of the filenames mentioned in the first link were referenced in Process Monitor throughout installation.

**Comment by SkyNinja** [comment]
No, not so far. I did [run it in Any Run](https://app.any.run/tasks/e9bad183-f80b-4b60-a559-f9df87c26cc3) as far as possible without an actual copy of SADX and got no abnormal behavior up to that point.

**Comment by PiKeyAr** [comment]
Okay. Thanks for testing. In that case I'm interested to know why you think it isn't a false positive.

**Comment by SkyNinja** [comment]
Admittedly that was before I found a sandboxing site I could use, but still, what gave me pause is the sheer depth of the detection. The fact that I seemed to pull this `541c6aa57ddd7da0c6902aa1e92155eb.virus` thing from the installer, execute it itself, and have it pull down so much nasty stuff with such winding connections, even leading back to similar but entirely distinct ".virus" executables. I've never seen a false positive go quite so deep or seem this... I guess consistent is the best word.

**Comment by PiKeyAr** [comment]
>The fact that I seemed to pull this 541c6aa57ddd7da0c6902aa1e92155eb.virus thing from the installer


You seemed to do that? How? Do you have that file, extracted from the installer's executable? If so, it's possible that your machine is infected, and the installer was among the infected files.


Let's assume the installer EXE actually has that virus. According to the VirusTotal page, `541c6aa57ddd7da0c6902aa1e92155eb.virus` is 435.72KB. The Mod Installer's EXE file (as of March 3rd, 2022) is 3 144 743 bytes. I don't know how VirusTotal detects "bundled" files, but if you open sadx_setup.exe in 7-Zip File Manager, it will show you all files that are included in it at compile time. Among those are SaveTransfer, 7-Zip, some graphics and various NSIS plugins used by the installer script. You can verify that there isn't a `541c6aa57ddd7da0c6902aa1e92155eb.virus` or any file of a similar size. The compressed size of all those items is 2 828 879 bytes, according to 7-Zip. That leaves 315 864 bytes for the potential "virus", which is already quite a bit smaller than `541c6aa57ddd7da0c6902aa1e92155eb.virus` is supposed to be according to VirusTotal. But there's also the NSIS stub (~40KB), the UI forms (~6KB), the icon (~56KB) and the actual compiled script and translations. So there just isn't any space to store another ~400KB file in it.


If you have any actual proof of malicious behavior in the installer, please post it. Until then, I'm going to mark this as a false positive.

**Comment by PiKeyAr** [comment]
Here's a compile log from Venis IX/NSIS just in case:
```
Processing config: C:\Program Files (x86)\NSIS\nsisconf.nsh
Processing script file: "D:\GitHub\sadx-mod-installer\Script.nsi" (ACP)

Processed 1 file, writing output (x86-unicode):

Output: "D:\GitHub\sadx-mod-installer\sadx_setup.exe"
Install: 18 pages (1152 bytes), 38 sections (4 required) (78736 bytes), 6549 instructions (183372 bytes), 4099 strings (1485676 bytes), 8 language tables (13488 bytes).
Datablock optimizer saved 389645 bytes (~11.7%).

Using zlib compression.

EXE header size:               99328 / 39936 bytes
Install code:                 216416 / 1762936 bytes
Install data:                2828995 / 3766530 bytes
CRC (0xEE29219B):                  4 / 4 bytes

Total size:                  3144743 / 5569406 bytes (56.4%)
```

**Comment by SkyNinja** [comment]
> > The fact that I seemed to pull this 541c6aa57ddd7da0c6902aa1e92155eb.virus thing from the installer
> 
> You seemed to do that? How? Do you have that file, extracted from the installer's executable? If so, it's possible that your machine is infected, and the installer was among the infected files.

That was a typo, meant to say 'it' as in VirusTotal. That said, it couldn't be the case that it was an infection from my system anyway, as it had been uploaded previously by someone else, and the hash of my freshly downloaded copy matched that previous upload.
